tests/unit/s2n_resume_test.c

```diff
@@ -163,6 +169,143 @@ int main(int argc, char **argv)
+            EXPECT_MEMCPY_SUCCESS(conn->client_ticket.data, client_ticket, sizeof(client_ticket));
+
+            EXPECT_OK(s2n_client_deserialize_session_state(conn, &ticket_stuffer));
+
+            struct s2n_psk *psk = NULL;
+            EXPECT_OK(s2n_array_get(&conn->psk_params.psk_list, 0, (void**) &psk));
+            EXPECT_NOT_NULL(psk);
+
+            EXPECT_EQUAL(psk->type, S2N_PSK_TYPE_RESUMPTION);
+            S2N_BLOB_EXPECT_EQUAL(psk->identity, conn->client_ticket);
+
+            EXPECT_EQUAL(psk->secret.size, SECRET_LEN);
+            uint8_t secret[] = { SECRET };
+            EXPECT_BYTEARRAY_EQUAL(psk->secret.data, secret, sizeof(secret));
+
+            EXPECT_EQUAL(psk->hmac_alg, S2N_HMAC_SHA256);
+            
+            uint8_t ticket_age_add[] = { TICKET_AGE_ADD };
+            uint32_t expected_ticket_age_add = ticket_age_add[0] | (ticket_age_add[1] << 8) | \
+                                              (ticket_age_add[2] << 16) | (ticket_age_add[3] << 24);
+            EXPECT_EQUAL(psk->obfuscated_ticket_age, expected_ticket_age_add);
+
+            uint8_t ticket_issue_time[] = { TICKET_ISSUE_TIME };
+            struct s2n_blob ticket_issue_time_blob = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&ticket_issue_time_blob, ticket_issue_time, sizeof(ticket_issue_time)));
+            
+            struct s2n_stuffer ticket_issue_time_stuffer = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_init(&ticket_issue_time_stuffer, &ticket_issue_time_blob));
+            EXPECT_SUCCESS(s2n_stuffer_skip_write(&ticket_issue_time_stuffer, sizeof(ticket_issue_time)));
+            uint64_t expected_ticket_issue_time = 0;
+            EXPECT_SUCCESS(s2n_stuffer_read_uint64(&ticket_issue_time_stuffer, &expected_ticket_issue_time));
+            EXPECT_EQUAL(psk->ticket_issue_time, expected_ticket_issue_time);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+        }
+
+        /* Functional test: The TLS1.3 client can deserialize what it serializes */
+        {
+            struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+            EXPECT_NOT_NULL(conn);
+
+            conn->actual_protocol_version = S2N_TLS13;
+            conn->secure.cipher_suite = &s2n_tls13_aes_256_gcm_sha384;
+            DEFER_CLEANUP(struct s2n_stuffer stuffer = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
+
+            struct s2n_ticket_fields ticket_fields = { .ticket_age_add = 0, .session_secret = test_session_secret };
+
+            /* Initialize client ticket */
+            uint8_t client_ticket[] = { CLIENT_TICKET };
+            EXPECT_SUCCESS(s2n_realloc(&conn->client_ticket, sizeof(client_ticket)));
+            EXPECT_MEMCPY_SUCCESS(conn->client_ticket.data, client_ticket, sizeof(client_ticket));
+
+            EXPECT_OK(s2n_serialize_resumption_state(conn, &ticket_fields, &stuffer));
+            EXPECT_OK(s2n_client_deserialize_session_state(conn, &stuffer));
+
```

Comment:
You should probably verify the PSK that results from this matches what you expect.